### ISSUE TITLE
CairoMakie fails on zeros heatmap

### ISSUE THREAD

**Issue by rongcuid** [JuliaPlots/Makie.jl#1748]
The error is:

```
ERROR: Looking up a non-finite or NaN value in a colormap is undefined.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] interpolated_getindex(cmap::Vector{ColorTypes.RGB{Float32}}, i01::Float64)
    @ Makie ~/.julia/packages/Makie/vwpRo/src/colorsampler.jl:47
  [3] interpolated_getindex
    @ ~/.julia/packages/Makie/vwpRo/src/colorsampler.jl:37 [inlined]
  [4] get_rgba_pixel(pixel::Float32, colormap::Vector{ColorTypes.RGB{Float32}}, colorrange::Vec{2, Float32}, nan_color::ColorTypes.RGBA{Float32}, lowclip::Nothing, highclip::Nothing)
    @ CairoMakie ~/.julia/packages/CairoMakie/JAJ5G/src/utils.jl:145
  [5] (::CairoMakie.var"#14#15"{Vec{2, Float32}, Vector{ColorTypes.RGB{Float32}}})(pixel::Float32)
    @ CairoMakie ./none:0
  [6] iterate
    @ ./generator.jl:47 [inlined]
  [7] collect(itr::Base.Generator{Matrix{Float32}, CairoMakie.var"#14#15"{Vec{2, Float32}, Vector{ColorTypes.RGB{Float32}}}})
    @ Base ./array.jl:724
  [8] to_rgba_image(img::Matrix{Float32}, attributes::Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}})
    @ CairoMakie ~/.julia/packages/CairoMakie/JAJ5G/src/utils.jl:130
  [9] to_cairo_image(img::Matrix{Float32}, attributes::Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}})
    @ CairoMakie ~/.julia/packages/CairoMakie/JAJ5G/src/utils.jl:120
 [10] draw_atomic(scene::Scene, screen::CairoMakie.CairoScreen{Cairo.CairoSurfaceBase{UInt32}}, primitive::Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}})
    @ CairoMakie ~/.julia/packages/CairoMakie/JAJ5G/src/primitives.jl:555
 [11] draw_plot(scene::Scene, screen::CairoMakie.CairoScreen{Cairo.CairoSurfaceBase{UInt32}}, primitive::Heatmap{Tuple{Vector{Float32}, Vector{Float32}, Matrix{Float32}}})
    @ CairoMakie ~/.julia/packages/CairoMakie/JAJ5G/src/infrastructure.jl:230
 [12] cairo_draw(screen::CairoMakie.CairoScreen{Cairo.CairoSurfaceBase{UInt32}}, scene::Scene)
    @ CairoMakie ~/.julia/packages/CairoMakie/JAJ5G/src/infrastructure.jl:172
 [13] backend_show(x::CairoMakie.CairoBackend, io::IOContext{Base64.Base64EncodePipe}, #unused#::MIME{Symbol("image/png")}, scene::Scene)
    @ CairoMakie ~/.julia/packages/CairoMakie/JAJ5G/src/infrastructure.jl:335
 [14] show(io::Base64.Base64EncodePipe, m::MIME{Symbol("image/png")}, scene::Scene)
    @ Makie ~/.julia/packages/Makie/vwpRo/src/display.jl:109
 [15] base64encode(::Function, ::MIME{Symbol("image/png")}, ::Vararg{Any}; context::Nothing)
    @ Base64 /opt/julia/julia-1.7.2/share/julia/stdlib/v1.7/Base64/src/encode.jl:209
 [16] _binstringmime
    @ /opt/julia/julia-1.7.2/share/julia/stdlib/v1.7/Base64/src/Base64.jl:46 [inlined]
 [17] stringmime(m::MIME{Symbol("image/png")}, x::Scene; context::Nothing)
    @ Base64 /opt/julia/julia-1.7.2/share/julia/stdlib/v1.7/Base64/src/Base64.jl:43
 [18] stringmime
    @ /opt/julia/julia-1.7.2/share/julia/stdlib/v1.7/Base64/src/Base64.jl:43 [inlined]
 [19] display(d::VSCodeServer.InlineDisplay, m::MIME{Symbol("image/png")}, x::Scene)
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/display.jl:53
 [20] display(d::VSCodeServer.InlineDisplay, mime::String, x::Any)
    @ Base.Multimedia ./multimedia.jl:216
 [21] display(d::VSCodeServer.InlineDisplay, x::Scene)
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/display.jl:156
 [22] display(x::Any)
    @ Base.Multimedia ./multimedia.jl:328
 [23] display(scene::Scene)
    @ Makie ~/.julia/packages/Makie/vwpRo/src/display.jl:57
 [24] #display#917
    @ ~/.julia/packages/Makie/vwpRo/src/display.jl:53 [inlined]
 [25] display
    @ ~/.julia/packages/Makie/vwpRo/src/display.jl:53 [inlined]
 [26] #display#916
    @ ~/.julia/packages/Makie/vwpRo/src/display.jl:52 [inlined]
 [27] display(fap::Makie.FigureAxisPlot)
    @ Makie ~/.julia/packages/Makie/vwpRo/src/display.jl:52
 [28] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [29] invokelatest
    @ ./essentials.jl:714 [inlined]
 [30] (::VSCodeServer.var"#60#64"{Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/eval.jl:176
 [31] withpath(f::VSCodeServer.var"#60#64"{Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams}, path::String)
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/repl.jl:185
 [32] (::VSCodeServer.var"#59#63"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/eval.jl:153
 [33] hideprompt(f::VSCodeServer.var"#59#63"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/repl.jl:36
 [34] (::VSCodeServer.var"#58#62"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/eval.jl:124
 [35] with_logstate(f::Function, logstate::Any)
    @ Base.CoreLogging ./logging.jl:511
 [36] with_logger
    @ ./logging.jl:623 [inlined]
 [37] (::VSCodeServer.var"#57#61"{VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/eval.jl:201
 [38] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [39] invokelatest(::Any)
    @ Base ./essentials.jl:714
 [40] macro expansion
    @ ~/.vscode/extensions/julialang.language-julia-1.5.11/scripts/packages/VSCodeServer/src/eval.jl:34 [inlined]
 [41] (::VSCodeServer.var"#55#56")()
    @ VSCodeServer ./task.jl:423
```

Minimal example:

```julia
using CairoMakie

##

m = zeros(10, 10)

##

ax, hm = heatmap(m)
```

Might be related to #1738.

**Comment by rongcuid** [comment]
Seems to be fixed from 0.16.3 -> 0.16.6. I need to check whether this is really working.

**Comment by rongcuid** [comment]
I confirm that the problem has been fixed in `0.16.6`.
